syntax = "proto2";

package GuindexAlertsIf;

// Message to send to AlertsServer when new price is added
message GuinnessCreateAlertRequest {
    required string pub          = 1;
    required string county       = 2;
    required string price        = 3;
    required string username     = 4;
    required string creationDate = 5;
    required bool   approved     = 6;
}

// Message to send to AlertsServer when pending Guinness create decision is made
message GuinnessPendingCreateDecisionAlertRequest {
    required string pub          = 1;
    required string county       = 2;
    required string price        = 3;
    required uint32 userId       = 4;
    required string creationDate = 5;
    required bool   approved     = 6;
    optional string reason       = 7;
}

// Message to send to AlertsServer when new Pub is added
message PubCreateAlertRequest {
    required string pub          = 1;
    required string county       = 2;
    required string latitude     = 3;
    required string longitude    = 4;
    required string username     = 5;
    required string creationDate = 6;
    required bool   approved     = 7;
}

// Message to send to AlertsServer when Pub is patched
message PubPatchAlertRequest {
    required string pub           = 1;
    required string county        = 2;
    required string username      = 3;
    required string changedFields = 4; // TODO Can we add constraints so this is encoded correctly?
    required string creationDate  = 5;
    required bool   approved      = 6;
}

// Message to send to AlertsServer when pending Pub create decision is made
message PubPendingCreateDecisionAlertRequest {
    required string pub          = 1;
    required string county       = 2;
    required string latitude     = 3;
    required string longitude    = 4;
    required uint32 userId       = 5;
    required string creationDate = 6;
    required bool   approved     = 7;
    optional string reason       = 8;
}

// Message to send to AlertsServer when pending Pub patch decision is made
message PubPendingPatchDecisionAlertRequest {
    required string pub           = 1;
    required string county        = 2;
    required uint32 userId        = 3;
    required string changedFields = 4; // TODO Can we add constraints so this is encoded correctly?
    required string creationDate  = 5;
    required bool   approved      = 6;
    optional string reason        = 7;
}

message GuindexAlertsIfMessage {
    optional GuinnessCreateAlertRequest                guinnessCreateAlertRequest                = 1;
    optional GuinnessPendingCreateDecisionAlertRequest guinnessPendingCreateDecisionAlertRequest = 2;
    optional PubCreateAlertRequest                     pubCreateAlertRequest                     = 3;
    optional PubPatchAlertRequest                      pubPatchAlertRequest                      = 4;
    optional PubPendingCreateDecisionAlertRequest      pubPendingCreateDecisionAlertRequest      = 5;
    optional PubPendingPatchDecisionAlertRequest       pubPendingPatchDecisionAlertRequest       = 6;
}
